### ISSUE TITLE
Testing iframes with Playwright/axe-core

### ISSUE THREAD

**Issue by zuzanna-maria** [dequelabs/axe-core#4522]
### Product

axe-core

### Latest Version

- [X] I have tested the issue with the latest version of the product

#### Expectation
I am working on accessibility testing with Playwright and @axe-core, following the instructions [here](https://playwright.dev/docs/accessibility-testing). My page contains iframes and I would like to scan and analyze the content inside them.

#### Actual
When scanning the page object as per the instructions, the `incomplete` property of the `accessibilityScanResults` object contains this:

```
    {
      id: 'frame-tested',
      impact: 'critical',
      tags: [Array],
      description: 'Ensures <iframe> and <frame> elements contain the axe-core script',
      help: 'Frames should be tested with axe-core',
      helpUrl: 'https://dequeuniversity.com/rules/axe/4.9/frame-tested?application=playwright',
      nodes: [Array]
    }
```
The link provided under the helpUrl property says to "Ensure iframe and frame elements contain the axe-core script" but doesn't provide an example of how to achieve that.

#### How to Reproduce
When testing a page containing an iframe with code provided in the documentation:

`const accessibilityScanResults = await new AxeBuilder({ page }).analyze()`

the above issue occurs. it does not occur on pages without iframes. It happens even when trying to explicitly include the iframe as per the documentation:

`const accessibilityScanResults = await new AxeBuilder({ page }).include('#sbox-iframe').analyze()`

#### Additional context
Any thing else we should know about the issue?

**Comment by straker** [comment]
Thanks for the issue. Unfortunately there are serval reason why an iframe wouldn't be tested. If for some reason axe was prevented from loading into the iframe (for example due to CORS), if the iframe was not fully loaded or even lazy loaded (`loading="lazy"` and thus not loaded into view when the page loads), or if the iframe threw a JavaScript error during the `axe.run` execution. Without seeing the actual page it's hard to say what the issue is exactly. 

You could try adding a wait to the script just to see if allowing the page more time to load fixes the issue.

```js
import { timeout } from 'node:timers/promises'

await page.goto('https://your-site.com/')
await setTimeout(5000)
const accessibilityScanResults = await new AxeBuilder({ page }).analyze()
```

**Comment by zuzanna-maria** [comment]
Thanks for your reply! it is indeed likely the injection is being blocked by security measures. Could you clarify what kind of injection is being performed by axe-core, as the documentation seems to lack detail on this?

**Comment by Billy-Stroud** [comment]
I am getting the same issue for an iFrame in our Salesforce application while use the `@axe-core/playwright` package.  Is there a way to determine if the script isn't being loaded due to CORS?

**Comment by straker** [comment]
@Billy-Stroud Unfortunately at the moment there isn't a way to get at this information. We're in discussions to see if we could expose the actual error, but it won't be any time soon.

As a way to see what's going on, I have this custom script I use to inject axe into each frame on the page. You could try running it on the pages in question and hopefully see the same type of error that @axe-core/playwright would be running into.

```js
function inject(doc) {
  const script = doc.createElement('script');
  script.src = 'https://unpkg.com/axe-core@latest';
  doc.body.appendChild(script);

  doc.querySelectorAll('iframe, frame').forEach(frame => {
    frame.contentDocument && inject(frame.contentDocument)
  });
}

inject(document)
```

**Comment by Billy-Stroud** [comment]
@straker Thank you!  That allowed me to see the CORS error.  It would definitely be helpful to expose these errors in the future.

**Comment by kkarthik-1** [comment]
@WilcoFiers 
In our case, axe reported a low color-contrast violation on a <u> element inside the wysihtml5 iframe (e.g., text color #eee8df on a white background).
Detecting this requires axe to access the iframe DOM and compute styles, so axe must have run inside the iframe.

Eg:

`<iframe class="wysihtml5-sandbox" 
        allowtransparency="true" 
        frameborder="0" 
        width="0" 
        height="0" 
        marginwidth="0" 
        marginheight="0" 
        title="Rich text editor" 
        style="display: block; 
               background-color: rgb(255, 255, 255); 
               border-collapse: separate; 
               border-color: rgb(190, 194, 185); 
               border-style: solid; 
               border-width: 1px; 
               clear: none; 
               float: none; 
               margin: 0px; 
               outline: rgb(30, 40, 95) none 0px; 
               outline-offset: 0px; 
               padding: 6px 12px; 
               position: static; 
               inset: auto; 
               z-index: auto; 
               vertical-align: baseline; 
               text-align: start; 
               box-shadow: rgba(0, 0, 0, 0.075) 0px 1px 1px 0px inset; 
               border-radius: 0px; 
               width: 1027.75px; 
               height: 114px;">
  <html>
    <head>
      <meta charset="UTF-8">
      <link rel="stylesheet" href="http://localhost:3000/as.css">
      <style type="text/css">
        html { height: 100%; }
        body { 
          min-height: 100%; 
          padding: 0; 
          margin: 0; 
          margin-top: -1px; 
          padding-top: 1px; 
        }
        ._wysihtml5-temp { display: none; }
        body.placeholder { color: #a9a9a9 !important; }
        body[disabled] { 
          background-color: #eee !important; 
          color: #999 !important; 
          cursor: default !important; 
        }
        img:-moz-broken { 
          -moz-force-broken-image-icon: 1; 
          height: 24px; 
          width: 24px; 
        }
      </style>
    </head>
    <body marginwidth="0" 
          marginheight="0" 
          class="form-control wysihtml5-editor" 
          spellcheck="true" 
          style="background-color: rgb(255, 255, 255); 
                 color: rgb(30, 40, 95); 
                 cursor: text; 
                 font-family: &quot;Public Sans&quot;, Helvetica, Arial, sans-serif; 
                 font-size: 14px; 
                 font-style: normal; 
                 font-variant: normal; 
                 font-weight: 400; 
                 line-height: 20px; 
                 letter-spacing: normal; 
                 text-align: start; 
                 text-decoration: none; 
                 text-indent: 0px; 
                 text-rendering: auto; 
                 word-break: normal; 
                 overflow-wrap: break-word; 
                 word-spacing: 0px;" 
          contenteditable="true">
      Test<br>
      Second header<br>
      <u style="color: #eee8df;">test</u>
    </body>
  </html>
</iframe>
`

**Comment by beanscg** [comment]
Bean Labs took a public-source pass on this issue.

The narrowest useful next step looks like: Offer a public comment that narrows the keyboard/ARIA or screen-reader expectation and identifies the component, example, or docs page to verify.

Public evidence checked:
- GitHub direct issue read: open, unassigned, comments=8, labels=question.
- Repository metadata read: license=MPL-2.0; archived=false.
- Timeline read: no open linked pull-request event found in fetched timeline page.

A good acceptance check would be a narrow public change that directly exercises the issue's reported boundary.

No PR opened from this pass.
